pmd-core/src/main/java/net/sourceforge/pmd/RuleSetWriter.java

```diff
@@ -266,21 +265,11 @@ private Element createPropertiesElement(List<PropertyDescriptor<?>> propertyDesc
         Element propertiesElement = null;
         if (propertyDescriptors != null) {
 
-            for (PropertyDescriptor<?> propertyDescriptor : propertyDescriptors) { // For
-                // each
-                // provided
-                // PropertyDescriptor
-
-                if (propertyDescriptor instanceof PropertyDescriptorWrapper) { // Any
-                    // wrapper
-                    // property
-                    // needs
-                    // to
-                    // go
-                    // out
-                    // as
-                    // a
-                    // definition.
+            for (PropertyDescriptor<?> propertyDescriptor : propertyDescriptors) {
+                // For each provided PropertyDescriptor
+
+                if (propertyDescriptor instanceof PropertyDescriptorWrapper) {
```

Comment:
I had consider `boolean`, but I felt it lacked clarity to help me get my point across.  Welcome to implement as such of course.

I wasn't considering a change on immutability.

It's not clear to me the factories can be use to delineate these scenarios, currently they appear to be publicly available, so Rules could be using them.

Perhaps having all the current means default to INTERNAL, provide another new means on the factory for creating with EXTERNAL, and use that where ever wrappers were created?  This should prevent accidental misuse, and not limit flexibility.